### PR TITLE
Optimization fixes

### PR DESCRIPTION
**Pull request by l4l**

### CODE REVIEW

File: core/infra/config/peer_service_with_json.cpp [hunk 1]
```diff
@@ -180,28 +181,22 @@ bool PeerServiceConfig::isExistPublicKey( const std::string &publicKey ) {
...
-  for(const auto &node : peerList) {
-    if(node.isOK()) {
-        nodes.push_back(std::make_unique<peer::Node>(node.getIP(),
-                                                     node.getPublicKey(),
-                                                     node.getTrustScore()));
-    }
-  }
+  Nodes nodes(peerList.size());
+  std::copy_if(peerList.begin(), peerList.end(), nodes.begin(),
+               [](const auto &node) { return (node)->isOK(); });
```
Comment (motxx) [inline core/infra/config/peer_service_with_json.cpp]: Is this deep-copy or not? Previous code runs deep copy, but new code might run shallow copy, right?

Comment (l4l) [inline core/infra/config/peer_service_with_json.cpp]: This one doesn't perform copy at all, only pointers.
`Nodes` is `std::vector<std::shared_ptr<Node>>`

Comment (motxx) [inline core/infra/config/peer_service_with_json.cpp]: Thanks, but I don't know whether getPeerList() allows peerList's destructive change or not.

Comment (motxx) [inline core/infra/config/peer_service_with_json.cpp]: This solution is best, I think.
http://melpon.org/wandbox/permlink/6Hy9qbQZxLAdvB7D

File: core/infra/config/peer_service_with_json.cpp [hunk 2]
```diff
@@ -29,15 +29,16 @@ limitations under the License.
...
+
 using txbuilder::TransactionBuilder;
 using type_signatures::Update;
 using type_signatures::Add;
 using type_signatures::Remove;
 using type_signatures::Peer;
 using nlohmann::json;
 
-std::vector<peer::Node> PeerServiceConfig::peerList;
+Nodes PeerServiceConfig::peerList;
```
Comment (satellitex) [inline core/infra/config/peer_service_with_json.cpp]: I'm sorry but, context of `peer_service_with_json.hpp` has migrated to `core/service/peer_service.hpp`.

Comment (satellitex) [inline core/infra/config/peer_service_with_json.cpp]: I'm sorry but, context of `peer_service_with_json.hpp` has migrated to `core/service/peer_service.hpp`.

File: core/consensus/sumeragi.cpp [hunk 3]
```diff
@@ -159,7 +157,7 @@ namespace sumeragi {
         std::int64_t    commitedCount = 0;
         std::uint64_t   numValidatingPeers;
         std::string     myPublicKey;
-        std::deque<std::unique_ptr<peer::Node>> validatingPeers;
+        std::vector<std::shared_ptr<peer::Node>> validatingPeers;
```
Comment (satellitex) [inline core/consensus/sumeragi.cpp]: `sumeragi` doesn't has to have peerList. It is enough to use `validatinIps`( that is gotten `getIpList()` ).
But, It occures copy of string. It is worse efficiency than `getPeerList` after fixed.

File: core/infra/config/peer_service_with_json.cpp (hunk 2 above)
Comment (motxx) [inline core/infra/config/peer_service_with_json.cpp]: I'll try to solve conflicts.

Comment (l4l) [inline core/infra/config/peer_service_with_json.cpp]: nevermind, i can rebase against new dev branch

File: core/consensus/sumeragi.cpp (hunk 3 above)
Comment (l4l) [inline core/consensus/sumeragi.cpp]: there's no copying at all
I didn't really understand what do you mean

File: core/infra/config/peer_service_with_json.cpp (hunk 2 above)
Comment (motxx) [inline core/infra/config/peer_service_with_json.cpp]: oh, thank you very much.

**Pull request merged**